From 616e694cd10288810cb489d5bd9cd9fc9fd400fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Tue, 6 Dec 2016 09:22:57 +0200
Subject: [PATCH 286/508] crypto: caam/qi - fix OF node refcount (use after
 free)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

"dev_node" is used after the reference to it is dropped.
Fix this by no longer using the node reference for printing
crypto algorithms registration info, since dev_info already
provides enough context.

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/caamalg_qi.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/caamalg_qi.c b/drivers/crypto/caam/caamalg_qi.c
index 7e88a2e..1ee8065 100644
--- a/drivers/crypto/caam/caamalg_qi.c
+++ b/drivers/crypto/caam/caamalg_qi.c
@@ -2294,8 +2294,7 @@ static int __init caam_qi_algapi_init(void)
 	}
 
 	if (!list_empty(&alg_list))
-		dev_info(priv->qidev, "%s algorithms registered in /proc/crypto\n",
-			 (char *)of_get_property(dev_node, "compatible", NULL));
+		dev_info(priv->qidev, "algorithms registered in /proc/crypto\n");
 
 	return err;
 }
-- 
1.7.5.4

